core/local/index.js

```diff
@@ -121,12 +123,19 @@ class Local /*:: implements Reader, Writer */ {
   /** Create a readable stream for the given doc */
   async createReadStreamAsync(
     doc /*: SavedMetadata */
-  ) /*: Promise<stream.Readable> */ {
+  ) /*: Promise<ReadableWithSize> */ {
     const filePath = this.abspath(doc.path)
     return new Promise((resolve, reject) => {
       const contentStream = fse.createReadStream(filePath)
-      contentStream.on('open', () => resolve(contentStream))
-      contentStream.on('error', err => reject(err))
+      contentStream.on('error', reject)
+      contentStream.on('open', () => {
+        // Once the promise is resolved, it can't be rejected so we should not
+        // expect later stream errors to reject it and can thus remove the
+        // listener.
+        contentStream.off('error', reject)
+
+        resolve(streamUtils.withSize(contentStream, doc.size || 0))
```

Comment:
It's for Flow. `Metadata.size` is defined as optional so it could theoretically be `undefined` although we know for sure it's not at this point.
It's a data structure design flow but it's a lot of work to get it right so I've left this on the side until I actually work on the data structures themselves.